### PR TITLE
V7: Move context and breadcrumbs to private properties

### PR DESCRIPTION
**Pull request by bengourley**
Per the spec, context and breadcrumbs should be private.

### CODE REVIEW

File: packages/core/config.js [hunk 1]
```diff
@@ -88,9 +88,14 @@ module.exports.schema = {
...
   },
+  context: {
+    defaultValue: () => undefined,
+    message: 'should be a string',
+    validate: value => value === undefined || typeof value === 'string'
+  },
   user: {
     defaultValue: () => null,
-    message: '(object) user should be an object',
+    message: 'user should be an object',
```
Comment (djskinner) [inline packages/core/config.js]: Could remove 'user' here? I guess the context would be sufficient to know it was user? Also makes it consistent with the other messages and may save a byte if we're lucky!

File: packages/core/client.js [hunk 2]
```diff
@@ -82,6 +79,14 @@ class Client {
     return metadataDelegate.clear(this._metadata, section, key)
   }
 
+  getContext () {
+    return this._context
+  }
+
+  setContext (c) {
+    this._context = c
+  }
```
Comment (djskinner) [inline packages/core/client.js]: Do these need adding to the public interface typedefs?

Comment (bengourley) [reply packages/core/client.js]: yeah – I was in the middle of a big types refactor when I realised this change was needed, so they are in #682 to save a conflict!

File: packages/core/config.js (hunk 1 above)
Comment (bengourley) [reply packages/core/config.js]: 6 bytes after minification!

**Pull request merged**